Midichlorians/Dev/Beta2/MidichloriansDataDictionary.h

```diff
@@ -0,0 +1,142 @@
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#ifndef __cplusplus
+// If you get this error building an Objective-C project, #include neither this file nor "MidichloriansImpl.h".
+// Use "Midichlorians.h" instead.
+// It wraps the inner, platform-independent, event tracking code, and allows client code to stay `.m`.
+#error "This C++ header should be `#include`-d or `#import`-ed from an `.mm`, not an `.m` source file."
+#endif
+
+#ifndef CURRENT_MIDICHLORIANS_DATA_DICTIONARY_H
+#define CURRENT_MIDICHLORIANS_DATA_DICTIONARY_H
+
+#include "../../../TypeSystem/struct.h"
+
+#ifdef COMPILE_MIDICHLORIANS_DATA_DICTIONARY_FOR_IOS_CLIENT
+#import <Foundation/Foundation.h>
+#import <CoreLocation/CoreLocation.h>
+#endif  // COMPILE_MIDICHLORIANS_DATA_DICTIONARY_FOR_IOS_CLIENT
+
+namespace current {
+namespace midichlorians {
+
+// Base event.
+CURRENT_STRUCT(MidichloriansBaseEvent) {
+  CURRENT_FIELD(user_ms, std::chrono::milliseconds);
+  CURRENT_FIELD(device_id, std::string);
+  CURRENT_FIELD(client_id, std::string);
+};
+
+// iOS-specific base event.
+CURRENT_STRUCT(iOSBaseEvent, MidichloriansBaseEvent) {
+  // TODO: do we need this field?
+  CURRENT_FIELD(description, std::string);
+};
+
+// iOS event denoting the very first application launch.
+// Emitted once during Midichlorians initialization.
+CURRENT_STRUCT(iOSFirstLaunchEvent, iOSBaseEvent){};
+
+// iOS event with the detailed application information.
+// Emitted once during Midichlorians initialization.
+CURRENT_STRUCT(iOSAppLaunchEvent, iOSBaseEvent) {
+  CURRENT_FIELD(binary_version, std::string);
+  CURRENT_FIELD(cf_version, std::string);
+  CURRENT_FIELD(app_install_time, uint64_t);
+  CURRENT_FIELD(app_update_time, uint64_t);
+  CURRENT_DEFAULT_CONSTRUCTOR(iOSAppLaunchEvent) {}
+  CURRENT_CONSTRUCTOR(iOSAppLaunchEvent)(
+      const std::string& cf_version, uint64_t app_install_time, uint64_t app_update_time)
+      : binary_version(__DATE__ " " __TIME__),
+        cf_version(cf_version),
+        app_install_time(app_install_time),
+        app_update_time(app_update_time) {
+    iOSBaseEvent::description = "*FinishLaunchingWithOptions";
+  }
+};
+
+// iOS event with the detailed device information.
+// Emitted once during Midichlorians initialization.
+CURRENT_STRUCT(iOSDeviceInfo, iOSBaseEvent) { CURRENT_FIELD(info, (std::map<std::string, std::string>)); };
```

Comment:
The iOS event collection part will likely not be compiled on Windows, but the analytics on those might well. Why not?